### ISSUE TITLE
Enable Souffle for 32bit platform

### ISSUE THREAD

**Issue by mmcgr** [souffle-lang/souffle#543]
Tests with a 32bit install of Ubuntu 16.04, using gcc 5.4/7 and openmp enabled, hang in the parallel section of trie_test.cpp. Commenting out the insert() inside the pfor avoids the hang. Removing the #pragma omp parallel also avoids the hang.

Some of the evaluation tests that use -c also hang.

**Comment by b-scholz** [comment]
Let's rephrase this as a feature request. I don't see an urgent need to compile/run Souffle on 32bit systems. However, if an urgent need arises, let's port the system to 32bit.

**Comment by mmcgr** [comment]
I'm closing this as we have no plans to spend resources making souffle work on 32 bit unless there's demand. A workaround is to configure without openmp enabled.

This may be re-opened if the demand arises in future.
